test/uk/gov/hmrc/tai/connectors/HttpHandlerSpec.scala

```diff
@@ -46,286 +59,281 @@ class HttpHandlerSpec extends BaseSpec with BeforeAndAfterEach {
+
+      server.stubFor(
+        get(anyUrl())
+          .willReturn(aResponse().withStatus(UNAUTHORIZED).withBody("unauthorized")))
+
+      val responseFuture = httpHandler.getFromApiV2(testUrl)
+      val ex = the[UnauthorizedException] thrownBy Await.result(responseFuture, 5 seconds)
+      ex.message must include("unauthorized")
+
     }
 
-    "result in an UnauthorisedException" when {
-      "when an Unauthorised response is received from the http get call" in {
-        val unauthorisedResponse =
-          HttpResponse(UNAUTHORIZED, Some(JsString("unauthorised response")), Map("ETag" -> Seq("34")))
+    "should return a HttpException when unknown response" in {
+
+      server.stubFor(
+        get(anyUrl())
+          .willReturn(aResponse().withStatus(IM_A_TEAPOT).withBody("unknown response")))
+
+      val responseFuture = httpHandler.getFromApiV2(testUrl)
+      val ex = the[Upstream4xxResponse] thrownBy Await.result(responseFuture, 5 seconds)
+      ex.message must include("unknown response")
 
-        when(http.GET[HttpResponse](any())(any(), any(), any())).thenReturn(Future.successful(unauthorisedResponse))
-        val responseFuture = sut.getFromApiV2("")
-        val ex = the[HttpException] thrownBy Await.result(responseFuture, 5 seconds)
-        ex.message mustBe "\"unauthorised response\""
-      }
     }
+
   }
 
-  "putToApi" should {
+  "putToApi" must {
+
     "return OK" in {
-      when(http.PUT[DateRequest, HttpResponse](any(), any(), any())(any(), any(), any(), any()))
-        .thenReturn(Future.successful(HttpResponse(OK)))
 
-      val result = Await.result(sut.putToApi[DateRequest]("", DateRequest(LocalDate.now())), 5.seconds)
+      server.stubFor(
+        put(anyUrl())
+          .willReturn(aResponse().withStatus(OK)))
+
+      val result = Await.result(httpHandler.putToApi[DateRequest](testUrl, DateRequest(LocalDate.now())), 5.seconds)
 
       result.status mustBe OK
 
     }
 
-    "return Not Found exception" in {
-      when(http.PUT[DateRequest, HttpResponse](any(), any(), any())(any(), any(), any(), any()))
-        .thenReturn(Future.successful(HttpResponse(NOT_FOUND)))
+    "should return a NotFoundException when NOT_FOUND response" in {
+
+      server.stubFor(
+        put(anyUrl())
+          .willReturn(aResponse().withStatus(NOT_FOUND)))
 
       val result = the[NotFoundException] thrownBy Await
-        .result(sut.putToApi[DateRequest]("", DateRequest(LocalDate.now())), 5.seconds)
+        .result(httpHandler.putToApi[DateRequest](testUrl, DateRequest(LocalDate.now())), 5.seconds)
 
       result.responseCode mustBe NOT_FOUND
+
     }
 
-    "return Internal Server exception" in {
-      when(http.PUT[DateRequest, HttpResponse](any(), any(), any())(any(), any(), any(), any()))
-        .thenReturn(Future.successful(HttpResponse(INTERNAL_SERVER_ERROR)))
+    "should return a InternalServerException when INTERNAL_SERVER_ERROR response" in {
```

Comment:
This code is throwing a Upstream5xxResponse not returning a HTTP code

Reply:
Changed name to Upstream5xxResponse